.github/workflows/ci.yml

```diff
@@ -56,34 +56,40 @@ jobs:
         run: make check-fmt
 
   tests:
-    name: ${{ matrix.make.name }} (${{ matrix.os }})
+    name: ${{ matrix.task.name }} (${{ matrix.os }})
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest]
         rust: [stable]
-        make:
+        task:
           - name: Clippy
-            task: "check-clippy"
-          - name: Unit tests
-            task: "build-all-test run-all-unit-test"
+            run: make check-clippy
+            publish: false
           - name: Doc tests
-            task: "run-all-doc-test"
+            run: make run-all-doc-test
+            publish: false
+          - name: Unit tests
+            run: make build-all-test run-all-unit-test
+            publish: true
         include:
           - os: ubuntu-latest
             sccache-path: /home/runner/.cache/sccache
+            target: x86_64-unknown-linux-musl
```

Comment:
We cannot add the targets directly to the matrix because it would mismatch the targets with the build hosts.

For example, if we had the following matrix:

```yaml
matrix:
  os: [ubuntu-latest, macos-latest]
  ...
  target: [x86_64-unknown-linux-musl, x86_64-apple-darwin]
```

Then we would get jobs with the following configurations:

```yaml
- os: ubuntu-latest
  target: x86_64-unknown-linux-musl
- os: ubuntu-latest
  target: x86_64-apple-darwin # This is mismatched with the host OS
- os: macos-latest
  target: x86_64-unknown-linux-musl # This is mismatched with the host OS
- os: macos-latest
  target: x86_64-apple-darwin
```

The include rule is the proper place for the target definition, since we can define which target gets built on which OS. I think it may be possible to add multiple targets for a given OS by defining multiple include rules, like the following:

```yaml
include:
- os: ubuntu-latest
  target: x86_64-unknown-linux-musl
- os: ubuntu-latest
  target: some-other-target-built-on-ubuntu-host
```